src/transformers/trainer.py

```diff
@@ -806,6 +806,7 @@ def get_train_dataloader(self) -> DataLoader:
             dataloader_params["sampler"] = self._get_train_sampler()
             dataloader_params["drop_last"] = self.args.dataloader_drop_last
             dataloader_params["worker_init_fn"] = seed_worker
+            dataloader_params["prefetch_factor"] = self.args.dataloader_prefetch_factor
```

Comment:
thanks for adding this @qmeeus. I was dealing with similar IterableDataset distributed loading issues recently as well :)

one quick question, is it supposed to add `prefectch_factor` configuration outside of this if branch, as this branch is for map-like dataset instead?